src/overlays/actors/ovl_En_Snowman/z_en_snowman.c

```diff
@@ -42,155 +73,1123 @@ const ActorInit En_Snowman_InitVars = {
+    Math_ScaledStepToS(&this->actor.shape.rot.y, Actor_YawToPoint(&this->actor, &this->combinePos), 0x1000);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+
+    if (this->combineState == EN_SNOWMAN_COMBINE_STATE_ACTIVE) {
+        if (this->collider.base.ocFlags1 & OC1_HIT) {
+            if ((this->collider.base.oc == this->actor.parent) &&
+                (parent->combineState == EN_SNOWMAN_COMBINE_STATE_ACTIVE)) {
+                if (this->actor.scale.x < this->actor.parent->scale.x) {
+                    EnSnowman_AbsorbEeno(this, parent);
+                } else {
+                    EnSnowman_AbsorbEeno(parent, this);
+                }
+            } else if ((this->collider.base.oc == this->actor.child) &&
+                       (child->combineState == EN_SNOWMAN_COMBINE_STATE_ACTIVE)) {
+                if (this->actor.scale.x < this->actor.child->scale.x) {
+                    EnSnowman_AbsorbEeno(this, child);
+                } else {
+                    EnSnowman_AbsorbEeno(child, this);
+                }
+            }
+        }
+
+        if (parent->combineState == EN_SNOWMAN_COMBINE_STATE_NO_ABSORPTION) {
+            EnSnowman_AbsorbEeno(parent, this);
+        }
+
+        if (child->combineState == EN_SNOWMAN_COMBINE_STATE_NO_ABSORPTION) {
+            EnSnowman_AbsorbEeno(child, this);
+        }
+    }
+
+    if ((this->combineTimer == 0) && (parent->work.targetScaleDuringCombine > 0.0f) &&
+        (child->work.targetScaleDuringCombine > 0.0f) && (this->work.targetScaleDuringCombine < 0.011f) &&
+        (this->combineState != EN_SNOWMAN_COMBINE_STATE_BEING_ABSORBED_OR_DONE)) {
+        this->combineState = EN_SNOWMAN_COMBINE_STATE_NO_ABSORPTION;
+        this->work.targetScaleDuringCombine = 0.0f;
+    }
+
+    if (Actor_XZDistanceToPoint(&this->actor, &this->combinePos) < 20.0f) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (Math_StepToF(&this->actor.scale.x, this->work.targetScaleDuringCombine, 0.0005f)) {
+        if (this->work.targetScaleDuringCombine < 0.01f) {
+            EnSnowman_SetupSplitDoNothing(this);
+        } else if (this->work.targetScaleDuringCombine > 0.018f) {
+            Actor_SetScale(&this->actor, 0.02f);
+            this->actor.params = EN_SNOWMAN_TYPE_LARGE;
+            this->actor.flags |= ACTOR_FLAG_400;
+            this->collider.base.ocFlags1 |= OC1_ON;
+            this->combineState = EN_SNOWMAN_COMBINE_STATE_BEING_ABSORBED_OR_DONE;
+            this->eenoScale = 2.0f;
+            EnSnowman_SetupMoveSnowPile(this);
+        }
+    }
+
+    this->actor.scale.y = this->actor.scale.x;
+    this->actor.scale.z = this->actor.scale.x;
+}
+
+void EnSnowman_UpdateDamage(EnSnowman* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        Actor_SetDropFlag(&this->actor, &this->collider.info);
+        if ((this->actor.colChkInfo.damageEffect != EN_SNOWMAN_DMGEFF_HOOKSHOT) ||
```

Comment:
I think we can actually name all the damage effects after the thing it does rather than the thing that does it: hookshot does nothing, Zora is Electric Stun, Fire Arrows is Melt?

Reply:
Hookshot is weird in that Link hooks onto and pulls himself towards big Eenos, but it damages small Eenos. I'm not sure what to even call that.

Reply:
That's handled by the collider rather than the damage effect system, right? But in that case it probably should stay as hookshot.

Reply:
Yeah, it's handled by the collider. I'll leave it as HOOKSHOT and change the rest to reflect what they do.